src/ape/api/transactions.py

```diff
@@ -40,6 +40,9 @@ class TransactionAPI(BaseInterfaceModel):
     # If left as None, will get set to the network's default required confirmations.
     required_confirmations: Optional[int] = Field(None, exclude=True)
 
+    # If true, will cause a TransactionError to be raised if the transaction fails.
+    _raise_on_fail: bool = False
```

Comment:
This value will need to remain `False` (or `None`) by default for this not to be a breaking change, but we did have a bit of discussion last week around potentially making it `True` during tests. Is that viable or would that cause its own issues?

Reply:
I think making it true during tests might make it work the same as people expect

Reply:
hmm, it must already raise on tests somewhere or else `ape.reverts()` wouldn't be working